crates/cubecl-wgpu/src/compute/server.rs

```diff
@@ -16,6 +16,72 @@ use cubecl_runtime::{
 use hashbrown::HashMap;
 use wgpu::{CommandEncoder, ComputePass, ComputePipeline, ShaderModuleDescriptor};
 
+#[cfg(not(target_family = "wasm"))]
+mod poll {
+    #[derive(Debug)]
+    pub struct WgpuPoll {
+        active_handle: std::sync::Arc<()>,
+        cancel_sender: std::sync::mpsc::Sender<()>,
+    }
+
+    impl WgpuPoll {
+        pub fn new(device: std::sync::Arc<wgpu::Device>) -> Self {
+            let active_handle = std::sync::Arc::new(());
+            let thread_check = active_handle.clone();
+
+            let (cancel_sender, cancel_receiver) = std::sync::mpsc::channel();
+            std::thread::spawn(move || loop {
+                // Check whether the WgpuPoll, this thread, and something else is holding
+                // a handle.
+                if std::sync::Arc::strong_count(&thread_check) > 2 {
+                    device.poll(wgpu::MaintainBase::Poll);
+                } else {
+                    // Do not cancel thread while someone still needs to poll.
+                    if cancel_receiver.try_recv().is_ok() {
+                        break;
+                    }
+                }
+                std::thread::yield_now();
+            });
```

Comment:
Not sure we should always spawn a new thread, could we actually have a pooling thread instead owned by the server?

Reply:
Well the server owns the WgpuPoll, there's one thread per server, not sure what you mean? 

Could have the spawn in the server and move it into WgpuPoll but idk, seperate type seems to work it to not copy more code to the spir-v server.

One thing I probably should do is park/unpark the thread when needed so it's not looping the entire time.

Reply:
I was confused, got it now thanks 😅